write.go

```diff
@@ -195,7 +195,11 @@ func (w *Writer) writeField(field interface{}) error {
 	case Literal:
 		return w.writeLiteral(field)
 	case []interface{}:
-		return w.writeList(field)
+		if len(field) == 0 {
```

Comment:
This is incorrect, an empty slice should be formatted as `()`

Reply:
that's what i thought, but it breaks the tests otherwise.

Reply:
`expected, _ := formatFields(test.fields)` will format incorrectly

Reply:
Not changing something shouldn't break tests that pass.

Reply:
except that those tests are failing because now that the envelope can have nil fields, the addresses change so `TestMessage_Format` and `TestEnveloppe_Format` will break.

Reply:
Should i create a new writer for the Envelope then ?

Reply:
hum my bad, i should just change the test data ...

Reply:
Can't we just fix `envelopeTests`? Basically replace `[]interface{}{}` with `nil`.

Reply:
should be all good now :)